service/src/java/org/apache/hive/service/server/HiveServer2.java

```diff
@@ -534,6 +455,102 @@ public synchronized void init(HiveConf hiveConf) {
     // Extra time for releasing the resources if timeout sets to 0
     ShutdownHookManager.addGracefulShutDownHook(() -> graceful_stop(),  timeout == 0 ? 30 : timeout);
   }
+  
+  private void addHAContextAttributes(HttpServer.Builder builder, HiveConf hiveConf) {
+    builder.setContextAttribute("hs2.isLeader", isLeader);
+    builder.setContextAttribute("hs2.failover.callback", new FailoverHandlerCallback(hs2HARegistry));
+    builder.setContextAttribute("hiveconf", hiveConf);
+  }
+  
+  private HttpServer.Builder initBuilder(String webHost, int port, String name, String contextPath, HiveConf hiveConf) throws IOException {
+    HttpServer.Builder builder = new HttpServer.Builder(name);
+    builder.setPort(port);
+    builder.setConf(hiveConf);
```

Comment:
wonder why we need both 
- `builder.setContextAttribute("hiveconf", hiveConf)`
- `builder.setConf(hiveConf)`?

Reply:
This was an existing code.

"hiveconf" was used in HS2Peers servlet. I refactored it to use `HttpServer.CONF_CONTEXT_ATTRIBUTE` that is set by HttpServer.Builder.